### DISCUSSION TITLE
[Request] System to block AFK Fishing

### DISCUSSION THREAD

**Discussion by TheWranger** [Zrips/CMI#371] [Feature Requests]
Implement a system to block users from automatically fishing while they are afk, threw the use of AFK machines or hacked clients. A simple way to do this would be to force players to move X blocks and change their angle slightly after X fishing sessions.
A configurable system like this is what I'm talking about:
https://www.spigotmc.org/resources/noafkfishing.29634/

**Comment by LogGits** [comment]
Great suggestion. Unfortunately cmi already has this feature. Look inside the afk section of the config relating to smart interact and make sure its enabled.

**Comment by TheWranger** [comment]
I don't think you understand what I mean.
`SmartInteractCheck` stops players from bypassing the afk check by fishing, but what I want is a system that stops players from using machines to continuously fish in the same spot without manually having to do anything.
@LogGits

**Comment by CloudeLecaw** [comment]
great i idea, i also use an AntiAFKFishing Plugin that Forces Players to move X blocks (config) to be able to use the fishingrod aggain, after they catched something
[NoAFKFishing+](https://www.spigotmc.org/resources/noafkfishing.29634/)
here u can see some config parts of (the relevant one in my eyes) may this min heal and remove hunger is not needed, but the
the first tree would be nice :-)
```
settings:
  angle-change-between-fishing: 0.3
  block-change-between-fishing: 1
  move-every-X-fishing: 3
```

for the langfile something would be needed like:
`  cannot-fish-same-place: You have to move a minimum of {block} blocks to be able to fish aggain.`
